### ISSUE TITLE
MAV_BATTERY_TYPE/MAV_BATTERY_FUNCTION enum collision

### ISSUE THREAD

**Issue by WickedShell** [mavlink/mavlink#697]
There is a ```MAV_BATTERY_TYPE_PAYLOAD``` stuck inside of the ```MAV_BATTERY_FUNCTION``` enum https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/common.xml#L2194 this should be ```MAV_BATTERY_FUNCTION_PAYLOAD```.

Is anyone using this/or would experience breakage if we fixed the name on the payload one to be ```MAV_BATTERY_FUNCTION_PAYLOAD```

This raises a generator question: If any param in the group isn't prefixed with the group name that should result in an error?

**Comment by DonLakeFlyer** [comment]
Yeah looks like a typo. The naming rule makes sense to me as well.

**Comment by LorenzMeier** [comment]
@WickedShell Could you send a PR fixing this?

**Comment by hamishwillee** [comment]
Fixed many years ago
